pkg/provisioner/ironic/ironic.go

```diff
@@ -963,6 +969,9 @@ func (p *ironicProvisioner) Provision(getUserData provisioner.UserDataSource) (r
 		return result, fmt.Errorf("no ironic node for host")
 	}
 
+	p.log.Info("Configuring hardware to host", "state", ironicNode.ProvisionState)
+	cleanSteps := p.buildAndValidateCleanSteps(hardwareConfig())
```

Comment:
Any information available from hardwareConfig() should also be available by looking at p.host directly, so this method should not need any arguments.

Reply:
Thanks. I removed  this `hardwareConfig()` and get the configuration directly from `p.host.Spec` as you said